pkg/runtime/framework/plugins/volcano/volcano_test.go

```diff
@@ -0,0 +1,399 @@
+							Volcano: &trainer.VolcanoPodGroupPolicySource{},
+						},
+					},
+				},
+				Scheduler: &runtime.Scheduler{},
+			},
+			existingPG: &volcanov1beta1.PodGroup{
+				ObjectMeta: metav1.ObjectMeta{Name: "job-exist-running", Namespace: "test-ns"},
+			},
+			expectPG:  nil,
+			expectErr: nil,
+		},
+		"PodGroup exists but trainjob suspended": {
+			trainJob: &trainer.TrainJob{
+				ObjectMeta: metav1.ObjectMeta{Name: "job-update", Namespace: "test-ns", UID: "2"},
+				Spec:       trainer.TrainJobSpec{Suspend: ptr.To(true)},
+			},
+			info: &runtime.Info{
+				TemplateSpec: baseInfo().TemplateSpec,
+				Annotations: map[string]string{
+					"scheduling.volcano.sh/queue-name": "q1",
+				},
+				RuntimePolicy: runtime.RuntimePolicy{
+					PodGroupPolicy: &trainer.PodGroupPolicy{
+						PodGroupPolicySource: trainer.PodGroupPolicySource{
+							Volcano: &trainer.VolcanoPodGroupPolicySource{
+								NetworkTopology: &volcanov1beta1.NetworkTopologySpec{
+									Mode:               volcanov1beta1.HardNetworkTopologyMode,
+									HighestTierAllowed: ptr.To(1),
+								},
+							},
+						},
+					},
+				},
+				Scheduler: &runtime.Scheduler{},
+			},
+			existingPG: &volcanov1beta1.PodGroup{ObjectMeta: metav1.ObjectMeta{Name: "job-update", Namespace: "test-ns"}},
+			expectPG: volcanov1beta1ac.PodGroup("job-update", "test-ns").
+				WithOwnerReferences(metav1ac.OwnerReference().
+					WithAPIVersion(trainer.GroupVersion.String()).
+					WithKind(trainer.TrainJobKind).
+					WithName("job-update").
+					WithUID(types.UID(strconv.Itoa(2))).
+					WithController(true).
+					WithBlockOwnerDeletion(true)).
+				WithSpec(volcanov1beta1ac.PodGroupSpec().
+					WithMinMember(5).
+					WithMinResources(corev1.ResourceList{
+						corev1.ResourceCPU:    resource.MustParse("2300m"),
+						corev1.ResourceMemory: resource.MustParse("3Gi"),
+					}).
+					WithQueue("q1").
+					WithPriorityClassName("high-priority").
+					WithNetworkTopology(&volcanov1beta1ac.NetworkTopologySpecApplyConfiguration{
+						Mode:               ptr.To(volcanov1beta1.HardNetworkTopologyMode),
+						HighestTierAllowed: ptr.To(1),
+					})),
+			expectErr: nil,
+		},
+		"Error when getting existing PodGroup": {
+			trainJob: &trainer.TrainJob{
+				ObjectMeta: metav1.ObjectMeta{Name: "job-error", Namespace: "test-ns", UID: "3"},
+				Spec:       trainer.TrainJobSpec{Suspend: ptr.To(false)},
+			},
+			info: &runtime.Info{
+				TemplateSpec: baseInfo().TemplateSpec,
+				RuntimePolicy: runtime.RuntimePolicy{
+					PodGroupPolicy: &trainer.PodGroupPolicy{
+						PodGroupPolicySource: trainer.PodGroupPolicySource{
+							Volcano: &trainer.VolcanoPodGroupPolicySource{},
+						},
+					},
+				},
+				Scheduler: &runtime.Scheduler{},
+			},
+			existingPG: nil,
+			expectPG:   nil,
+			expectErr:  mockGetError,
+		},
+	}
+
+	for name, c := range cases {
+		t.Run(name, func(t *testing.T) {
+			clientBuilder := fake.NewClientBuilder().WithScheme(scheme)
```

Comment:
Instead of `fake` lib, introducing https://github.com/kubeflow/trainer/blob/70fc13aa2f98ecf656263b070ab608e3f3efe64a/pkg/util/testing/client.go#L34